packages/block-editor/src/store/actions.js

```diff
@@ -484,14 +484,16 @@ export function insertBlock(
  * @param {?number}  index           Index at which block should be inserted.
  * @param {?string}  rootClientId    Optional root client ID of block list on which to insert.
  * @param {?boolean} updateSelection If true block selection will be updated.  If false, block selection will not change. Defaults to true.
+ * @param {?string}  patternName     Optional If inserted as a block pattern, the namespace of the block pattern.
  *
  *  @return {Object} Action object.
  */
 export function* insertBlocks(
 	blocks,
 	index,
 	rootClientId,
-	updateSelection = true
+	updateSelection = true,
+	patternName = false
```

Comment:
I wonder if a default of false for a string value is the best option here. Another option is to consider that if a string value is optional that it is 'undefined' if not set, which is the default behaviour if we don't explicitly set the default. Given that we have would have to check if the value is set whether it is false or undefined,  maybe the default default makes sense? This also follows the pattern for rootClientId which is also optional but doesn't set a default.

Reply:
Since this value is not directly used here and more of "meta" information. Some blocks can come from patterns, others from pasting... maybe we can have something more generif: `meta: { patternName: "something" }`

Reply:
Switched this to make use of undefined, and also as a generic meta object.